pkg/cloud/services/secretsmanager/helpers.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secretsmanager
+
+func splitBytes(data []byte, maxSize int, iterFunc func([]byte)) {
+	var split []byte
+	for len(data) >= maxSize {
+		split, data = data[:maxSize], data[maxSize:]
```

Comment:
Any specific reason to go this route vs using a buffer?

Reply:
No, but this is easier to read imo. We can change it up later if necessary, seems using a buffer is mostly a performance improvement?

Reply:
Personally, I find the following a bit easier to read and potentially a bit more approachable for new contributors:

```go
func splitBytes(data []byte, maxSize int, iterFunc func([]byte)) {
	buff := NewBuffer(data)
	for {
		chunk := buff.Next(maxSize)
		if len(chunk) == 0 {
			return
		}
		iterFunc(chunk)
	}
}
```

It's clear that we are returning once we have processed all the data and there is less chance of confusion around if we are handling boundaries properly.

Reply:
Sure, that is better! I’ll change it up later

Reply:
Swapped it in this PR.